src/test/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyRegistersServiceImplTest.java

```diff
@@ -0,0 +1,224 @@
+        assertNotNull(registers);
+        assertEquals(1, registers.size());
+        assertEquals("directors", registers.get("directors").getRegisterType());
+        assertEquals("/company/12345678/officers?register_view=true&register_type=directors",
+                registers.get("directors").getLinks().get("directors_register"));
+        verify(repository, times(1)).save(any(CompanyRegisters.class));
+    }
+
+    @Test
+    void testCreateCompanyRegistersWithMultipleRegisters() throws DataException {
+        setRegister("directors");
+        RegistersSpec secretariesRegister = new RegistersSpec();
+        secretariesRegister.setRegisterType("secretaries");
+        secretariesRegister.setRegisterMovedTo("Companies House");
+
+        companySpec.setRegisters(List.of(
+                companySpec.getRegisters().getFirst(),
+                secretariesRegister
+        ));
+
+        CompanyRegisters createdRegisters = service.create(companySpec);
+        assertEquals(2, createdRegisters.getRegisters().size());
+        assertNotNull(createdRegisters.getRegisters().get("directors"));
+        assertNotNull(createdRegisters.getRegisters().get("secretaries"));
+    }
+
+    @Test
+    void testGenerateRegisterLinks() throws DataException {
+        setRegister("directors");
+        CompanyRegisters createdRegisters = service.create(companySpec);
+        Map<String, String> links = createdRegisters.getRegisters()
+                .get("directors").getLinks();
+
+        assertNotNull(links);
+        assertEquals(1, links.size());
+        assertEquals("/company/12345678/officers?register_view=true&register_type=directors",
+                links.get("directors_register"));
+    }
+
+    @Test
+    void testGenerateRegisterLinksForSecretaries() throws DataException {
+        setRegister("secretaries");
+        CompanyRegisters createdRegisters = service.create(companySpec);
+        Map<String, String> links = createdRegisters.getRegisters()
+                .get("secretaries").getLinks();
+
+        assertNotNull(links);
+        assertEquals(1, links.size());
+        assertEquals("/company/12345678/officers?register_view=true&register_type=secretaries",
+                links.get("secretaries_register"));
+    }
+
+    @Test
+    void testGenerateRegisterLinksForPsc() throws DataException {
+        setRegister("persons-with-significant-control");
+        CompanyRegisters createdRegisters = service.create(companySpec);
+        Map<String, String> links = createdRegisters.getRegisters()
+                .get("persons_with_significant_control").getLinks();
+
+        assertNotNull(links);
+        assertEquals(1, links.size());
+        assertEquals("/company/12345678/persons-with-significant-control?register_view=true",
+                links.get("persons_with_significant_control_register"));
+    }
+
+    @Test
+    void testDeleteCompanyRegistersExists() {
```

Comment:
```suggestion
    void testDeletedCompanyRegistersExists() {
```
Is this better? I wasn't quite sure what these two tests are checking...